### PR TITLE
Add og_metadata to Dataset pages

### PR DESCRIPTION
**Pull request by adam3smith**
## Related Issues

- connects to https://github.com/IQSS/dataverse/issues/4894 

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [ ] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: src/main/webapp/dataset.xhtml [hunk 1]
```diff
@@ -39,6 +39,18 @@
                     <h:outputText value="#{DatasetPage.jsonLd}"/>
                 </script>
 	    </ui:define>
+      <ui:define name="og_header">
+      <meta property="og:title" content="#{DatasetPage.title}" />
+      <meta property="og:type" content="article" />
+      <meta property="og:url" content="#{DatasetPage.dataverseSiteUrl}/dataset.xhtml?persistentId=#{dataset.globalId}" />
+      <meta property="og:image" content="#{DatasetPage.thumbnailString}" />
+      <meta property="og:site_name" content="#{DatasetPage.publisher}" />
+      <meta property="og:description" content="#{DatasetPage.description}" />
```
Comment (adam3smith) [inline src/main/webapp/dataset.xhtml:48]: This should be limited to 150 chars, don't know how to do that.

File: src/main/webapp/dataset.xhtml [hunk 2]
```diff
@@ -39,6 +39,18 @@
                     <h:outputText value="#{DatasetPage.jsonLd}"/>
                 </script>
 	    </ui:define>
+      <ui:define name="og_header">
+      <meta property="og:title" content="#{DatasetPage.title}" />
+      <meta property="og:type" content="article" />
+      <meta property="og:url" content="#{DatasetPage.dataverseSiteUrl}/dataset.xhtml?persistentId=#{dataset.globalId}" />
+      <meta property="og:image" content="#{DatasetPage.thumbnailString}" />
```
Comment (adam3smith) [inline src/main/webapp/dataset.xhtml:46]: If this is empty, it should use the page favicon

**Comment by adam3smith** [comment]
@qqmyers If you could take a look at my two questions, see if the rest makes sense & merge if it does? No particular rush, but would be one more item from the roadmap.

**Comment by qqmyers** [comment]
@adam3smith - I merged and deployed to dev. I'll look at your questions there since I can edit the xhtml directly on dev without redeploying...

**Pull request merged**